### PR TITLE
[Messages] Change message duration to uint64

### PR DESCRIPTION
**Pull request by hasty**
This changes the duration on Messages to be uint64 to match changes in the spec.

### CODE REVIEW

File: examples/tv-app/tv-common/clusters/messages/MessagesManager.h [hunk 1]
```diff
@@ -110,7 +110,7 @@ struct CachedMessage
     const chip::app::Clusters::Messages::MessagePriorityEnum mPriority;
     const chip::BitMask<chip::app::Clusters::Messages::MessageControlBitmap> mMessageControl;
     const chip::app::DataModel::Nullable<uint32_t> mStartTime;
-    const chip::app::DataModel::Nullable<uint16_t> mDuration;
+    const chip::app::DataModel::Nullable<uint64_t> mDuration;
```
Comment (chrisdecenzo) [inline examples/tv-app/tv-common/clusters/messages/MessagesManager.h:113]: Line 129:
        const chip::app::DataModel::Nullable<uint64_t> & startTime, const chip::app::DataModel::Nullable<uint16_t> & duration,

File: examples/tv-app/android/java/MessagesManager.cpp [hunk 2]
```diff
@@ -186,7 +186,7 @@ CHIP_ERROR MessagesManager::HandleGetMessages(AttributeValueEncoder & aEncoder)
             jint jduration         = env->GetIntField(messageObject, durationField);
             if (jduration >= 0)
             {
-                message.duration = DataModel::Nullable<uint16_t>(static_cast<uint16_t>(jduration));
+                message.duration = DataModel::Nullable<uint16_t>(static_cast<uint64_t>(jduration));
```
Comment (chrisdecenzo) [inline examples/tv-app/android/java/MessagesManager.cpp:189]: other lines to change in this file due to data type change:

Line 77:
env->GetMethodID(managerClass, "presentMessages", "(Ljava/lang/String;IIJJLjava/lang/String;Ljava/util/HashMap;)Z");

Line 185:
jfieldID durationField = env->GetFieldID(messageClass, "duration", "J");
jlong jduration = env->GetLongField(messageObject, durationField);
if (jduration >= 0)
{
message.duration = DataModel::Nullable<uint64_t>(static_cast<uint64_t>(jduration));
}

Line 340:
jlong jduration = -1;
if (!duration.IsNull())
{
jduration = static_cast(duration.Value());
}

**Review by chrisdecenzo — CHANGES_REQUESTED** [review]

File: examples/tv-app/tv-common/clusters/messages/MessagesManager.h (hunk 1 above)
Comment (hasty) [reply examples/tv-app/tv-common/clusters/messages/MessagesManager.h:113]: Done.

File: examples/tv-app/android/java/MessagesManager.cpp (hunk 2 above)
Comment (hasty) [reply examples/tv-app/android/java/MessagesManager.cpp]: Done.

**Review by chrisdecenzo — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

File: data_model/clusters/Messages.xml [hunk 3]
```diff
@@ -216,7 +216,7 @@ Davis, CA 95616, USA
         <quality nullable="true"/>
         <mandatoryConform/>
       </field>
-      <field id="4" name="Duration" type="uint16" default="0">
+      <field id="4" name="Duration" type="uint64" default="0">
```
Comment (andy31415) [inline data_model/clusters/Messages.xml:219]: This is based on a scrape of the specification (which was done last week).

It is not meant for manual change. Also looking at my copy of the spec:

image

**Review by andy31415 — DISMISSED** [review]
@hasty  I do not see this change in the spec and this PR tries to manually change an automatically generated file.

Duration says is "in minutes" ... why do we want 64-bit there? Is about 45 days insufficient? even then probably 32-bit would be sufficient...

**Comment by andy31415** [comment]
> @hasty I do not see this change in the spec and this PR tries to manually change an automatically generated file.
> 
> Duration says is "in minutes" ... why do we want 64-bit there? Is about 45 days insufficient? even then probably 32-bit would be sufficient...

I see https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/8909 exists ... however that has changes-requested.

I feel uncomfortable allowing a SDK change before spec changes (very late). @robszewczyk what is the plan for the spec here? I would be ok if we synchronize them at the same time however right before 1.3 it seems dangerous to allow one to diverge before the other.

**Review by robszewczyk — APPROVED** [review]

**Pull request merged**